include/verilogAST.hpp

```diff
@@ -69,22 +69,53 @@ class NumericLiteral : public Expression {
   unsigned int size;  // default 32
   bool _signed;       // default false
   Radix radix;        // default decimal
+  // always generate size prefix (default false will not codegen size for 32
+  // bit literals)
+  bool always_codegen_size;
```

Comment:
you can just do `bool always_codegen_size = false;` to have it be a default, then only set it when it is specified (i.e. in the first constructor)

(same for the other defaults)